src/UCTSearch.cpp

```diff
@@ -45,16 +46,79 @@ UCTSearch::UCTSearch(GameState& g)
     m_root = std::make_unique<UCTNode>(FastBoard::PASS, 0.0f, 0.5f);
 }
 
+bool UCTSearch::advance_to_new_rootstate() {
+    if (!m_root || !m_last_rootstate) {
+        // No current state
+        return false;
+    }
+
+    if (m_rootstate.get_komi() != m_last_rootstate->get_komi()) {
+        return false;
+    }
+
+    auto depth =
+        (int) (m_rootstate.get_movenum() - m_last_rootstate->get_movenum());
```

Comment:
I believe this works but It's not as pretty as I'd like, a quick Google didn't suggest anything better.

Reply:
Agree, this kinda sucks in C++.